skyportal/handlers/api/reminder.py

```diff
@@ -0,0 +1,796 @@
+                }
+            )
+
+    @permissions(['Reminder'])
+    def post(self, associated_resource_type, resource_id):
+        """
+        ---
+        description: Post a reminder
+        tags:
+          - reminders
+        parameters:
+          - in: path
+            name: associated_resource_type
+            required: true
+            schema:
+              type: string
+              enum: [sources, spectrum, gcn_event]
+            description: |
+               What underlying data the reminder is on:
+               "source" or "spectrum" or "gcn_event".
+          - in: path
+            name: resource_id
+            required: true
+            schema:
+              type: string
+              enum: [sources, spectra, gcn_event]
+            description: |
+               The ID of the source or spectrum
+               that the reminder is posted to.
+               This would be a string for a source ID
+               or an integer for a spectrum.
+        requestBody:
+          content:
+            application/json:
+              schema:
+                type: object
+                properties:
+                  text:
+                    type: string
+                  group_ids:
+                    type: array
+                    items:
+                      type: integer
+                    description: |
+                      List of group IDs corresponding to which groups should be
+                      able to view reminder. Defaults to all of requesting user's
+                      groups.
+                required:
+                  - text
+        responses:
+          200:
+            content:
+              application/json:
+                schema:
+                  allOf:
+                    - $ref: '#/components/schemas/Success'
+                    - type: object
+                      properties:
+                        data:
+                          type: object
+                          properties:
+                            reminder_id:
+                              type: integer
+                              description: New reminder ID
+        """
+        data = self.get_json()
+
+        reminder_text = data.get("text")
+        next_reminder = data.get("next_reminder")
+        next_reminder = arrow.get(next_reminder).datetime.replace(tzinfo=None)
+        reminder_delay = data.get("reminder_delay", 0)
+        number_of_reminders = data.get("number_of_reminders", 1)
+        with self.Session() as session:
+            group_ids = data.pop('group_ids', None)
+            if not group_ids:
+                group_ids = [g.id for g in self.current_user.accessible_groups]
+            try:
+                groups = session.scalars(
+                    Group.select(self.current_user).where(Group.id.in_(group_ids))
+                ).all()
+            except AccessError:
+                return self.error('Could not find any accessible groups.', status=403)
```

Comment:
This should not be a try/except, but a test if the set of group IDs is the same as that given. 
And I would rephrase the error to "cannot find some of the requested groups`

Reply:
Done.